airbyte-cdk/bulk/core/load/src/integrationTest/kotlin/io/airbyte/cdk/load/mock_integration_test/MockBasicFunctionalityIntegrationTest.kt

```diff
@@ -246,8 +292,201 @@ class MockBasicFunctionalityIntegrationTest :
+            primaryKey = listOf(listOf("id")),
+            cursor = null,
+        )
+    }
+
+    @Test
+    open fun testNamespaceMappingDestinationNoPrefix() {
+        testNamespaceMapping(
+            NamespaceMappingConfig(namespaceDefinitionType = NamespaceDefinitionType.DESTINATION)
+        ) { _, mappedNamespace, unmappedName, mappedName ->
+            // For destination namespace mapping, the namespace should be the unmapped name.
+            assertNull(mappedNamespace)
+            assertEquals(unmappedName, mappedName)
+        }
+    }
+
+    @Test
+    open fun testNamespaceMappingDestinationWithPrefix() {
+        testNamespaceMapping(
+            NamespaceMappingConfig(
+                namespaceDefinitionType = NamespaceDefinitionType.DESTINATION,
+                streamPrefix = "prefix_",
+            )
+        ) { _, mappedNamespace, unmappedName, mappedName ->
+            // For destination namespace mapping, the namespace should be the unmapped name.
+            assertNull(mappedNamespace)
+            assertEquals("prefix_$unmappedName", mappedName)
+        }
+    }
+
+    @Test
+    open fun testNamespaceMappingSourceWithPrefix() {
+        testNamespaceMapping(
+            NamespaceMappingConfig(
+                namespaceDefinitionType = NamespaceDefinitionType.SOURCE,
+                streamPrefix = "prefix_",
+            )
+        ) { unmappedNamespace, mappedNamespace, unmappedName, mappedName ->
+            // For source namespace mapping, the namespace should be the unmapped namespace.
+            assertEquals(unmappedNamespace, mappedNamespace)
+            assertEquals("prefix_$unmappedName", mappedName)
+        }
+    }
+
+    @Test
+    open fun testNamespaceMappingCustomFormatNoPrefix() {
+        testNamespaceMapping(
+            NamespaceMappingConfig(
+                namespaceDefinitionType = NamespaceDefinitionType.CUSTOM_FORMAT,
+                namespaceFormat = "custom_\${SOURCE_NAMESPACE}_namespace",
+            )
+        ) { _, mappedNamespace, unmappedName, mappedName ->
+            // For custom namespace mapping, the namespace should be the custom format.
+            assertEquals("custom_${randomizedNamespace}_namespace", mappedNamespace)
+            assertEquals(unmappedName, mappedName)
+        }
+    }
+
+    @Test
+    open fun testNamespaceMappingCustomFormatNoMacroWithPrefix() {
+        testNamespaceMapping(
+            NamespaceMappingConfig(
+                namespaceDefinitionType = NamespaceDefinitionType.CUSTOM_FORMAT,
+                namespaceFormat = "custom_$randomizedNamespace",
+                streamPrefix = "prefix_",
+            )
+        ) { _, mappedNamespace, unmappedName, mappedName ->
+            // For custom namespace mapping, the namespace should be the custom format.
+            assertEquals("custom_${randomizedNamespace}", mappedNamespace)
+            assertEquals("prefix_$unmappedName", mappedName)
+        }
+    }
 }
+
```

Comment:
The two enabled test classes represent normal + speed mode. Speed mode has some disabled tests, see comments for details.